### PR TITLE
Add some info about cedta to importing vignette

### PR DESCRIPTION
**Pull request by MichaelChirico**
@jangorecki I forget if we're compiling the vignettes ourselves? Or if compilation is happening in CI?

Inspired by #2341

### CODE REVIEW

**Comment by jangorecki** [comment]
vignette is being compiled on R CMD build, but we do not push it to repo

File: vignettes/datatable-importing.Rmd [hunk 1]
```diff
@@ -148,12 +160,23 @@ my.write = function (x) {
   }
 }
 ```
-When using package as a suggested dependency you should not import it in NAMESPACE file, just mention it in DESCRIPTION file. You also have to use `data.table::` prefix when calling data.table functions because none of them are imported.
+
+When using a package as a suggested dependency, you should not `import` it in the `NAMESPACE` file. Just mention it in the `DESCRIPTION` file. You also _must_ use the `data.table::` prefix when calling `data.table` functions because none of them are imported.
+
+## `data.table` in `Imports:` but nothing exported
```
Comment (jangorecki) [inline vignettes/datatable-importing.Rmd]: "nothing imported" seems more appropriate, and I would remove `:` from `Imports:`

File: vignettes/datatable-importing.Rmd [hunk 2]
```diff
@@ -148,12 +160,23 @@ my.write = function (x) {
...
+
+Some users may prefer ([e.g.](https://github.com/Rdatatable/data.table/issues/2341)) may prefer to eschew using `importFrom` or `import` in their `NAMESPACE` file and instead use `data.table::` qualification on all internal code (of course keeping `data.table` under their `Imports:` in `DESCRIPTION`).
+
+In this case, the un-exported function `[.data.table` will revert to calling `[.data.frame` since `data.table` has no way of knowing that the parent package is aware it's attempting to make calls against the syntax of `data.table`'s query API (which could lead to unexpected errors).
```
Comment (jangorecki) [inline vignettes/datatable-importing.Rmd]: "unexpected errors" could be clarified, it means that API for data.frame and dt is different. Reader might got impression there are some non-handled, non-recognized (unexpected) errors occurring.

Comment (MichaelChirico) [reply vignettes/datatable-importing.Rmd]: Will do. Admittedly I'm flying a _bit_ by the seat of my pants here as I don't have _deep_ understanding of what's going on, just the basic idea.

File: vignettes/datatable-importing.Rmd (hunk 1 above)
Comment (MichaelChirico) [reply vignettes/datatable-importing.Rmd]: Yes, good

**Pull request merged**